storage/blockdevice/COMPONENT_SD/CMakeLists.txt

```
@@ -3,10 +3,10 @@
 
 target_sources(mbed-os
     PRIVATE
-        SDBlockDevice.cpp
+        source/SDBlockDevice.cpp
 )
 
 target_include_directories(mbed-os
     PUBLIC
-        ${CMAKE_CURRENT_SOURCE_DIR}
+        include/SD
```

```suggestion
        ${CMAKE_CURRENT_SOURCE_DIR}
        ${CMAKE_CURRENT_SOURCE_DIR}/include
        ${CMAKE_CURRENT_SOURCE_DIR}/include/SD
```
Just to be sure because some files include headers from different levels.
Also can you check that the other blockdevice components are correct? I did `QSPIF` and `FLASHIAP` because they were spotted when I tried to build Blinky for `NRF52840_DK`. But I did not see `SD`.

---

Fixed, I added include. I don't think we want to expose the root of the component if it is not needed. We should only include folder exposed by default.

---

I also fixed the rest of storage components, the same problem